chef-config/lib/chef-config/config.rb

```diff
@@ -857,7 +857,7 @@ def self.windows_home_path
       # Full URL to the endpoint that will receive our data. If nil, the
       # data collector will not run.
       # Ex: http://my-data-collector.mycompany.com/ingest
-      default :server_url,       nil
+      default(:server_url) { File.join(config_parent.chef_server_url, "/data_collector") }
```

Comment:
Chef server currently has a minimal proxy config for the data collector at `/data-collector`

https://github.com/chef/chef-server/blob/master/omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx_chef_api_lb.conf.erb#L48-L57